### DISCUSSION TITLE
Decrypting an encrypted string from DB

### DISCUSSION THREAD

**Discussion by fessor10** [adonisjs/core#2626] [Help] [answered]
I'm trying to decrypt an encrypted string from the database but it returns a different value from what was encrypted and saved to the DB. I'm using the adonis encryption for this.

Any help with this?

**Comment by RomainLanz** [comment, marked as answer]
Hey @fessor10! 👋 

You can use the `consume` and `prepare` [modifier](https://docs.adonisjs.com/reference/orm/decorators#column) for that.

```ts
import { BaseModel, column } from '@ioc:Adonis/Lucid/Orm'
import Encryption from '@ioc:Adonis/Core/Encryption'

class Model extends BaseModel {

  // ...
  
  @column({
    consume: (value?: Buffer) => (!value ? null : Encryption.decrypt(value.toString())),
    prepare: (value?: unknown) => (!value ? null : Encryption.encrypt(value)),
  })
  public secret: Record<string, string>
  
  // ...
  
}
```

**Reply by fessor10** [reply]
Hi @RomainLanz 

Thank you for the help ❤️
